File: CPP/Monitor.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;


class Monitor : public Komponen
{
private:
    int resolusi_x;
    int resolusi_y;
    string port_display;
public:
    Monitor()
    {


    }


    Monitor(int resolusi_x, int resolusi_y, string port_display, string merk, string nama) : Komponen(merk, nama)
    {
        this->resolusi_x = resolusi_x;
        this->resolusi_y = resolusi_y;
        this->port_display = port_display;
    }


    void setresolusi_x(int resolusi_x)
    {
        this->resolusi_x = resolusi_x;
    }
    
    void setresolusi_y(int resolusi_y)
    {
        this->resolusi_y = resolusi_y;
    }


    void setport_display(float port_display)
    {
        this->port_display = port_display;
    }


    int getresolusi_x()
    {
        return this->resolusi_x;
    }
    
    int getresolusi_y ()
    {
        return this->resolusi_y ;
    }


    string getport_display()
    {
        return this->port_display;
    }


    ~Monitor()
    {


    }
};

File: Python/Monitor.py
from Komponen import Komponen

class Monitor(Komponen):
    def __init__(self, resolusi_x=0, resolusi_y=0, port_display="", merk="", nama=""):
        super().__init__(merk, nama)
        self.resolusi_x = resolusi_x
        self.resolusi_y = resolusi_y
        self.port_display = port_display

    def set_resolusi_x(self, resolusi_x):
        self.resolusi_x = resolusi_x

    def set_resolusi_y(self, resolusi_y):
        self.resolusi_y = resolusi_y

    def set_port_display(self, port_display):
        self.port_display = port_display

    def get_resolusi_x(self):
        return self.resolusi_x

    def get_resolusi_y(self):
        return self.resolusi_y

    def get_port_display(self):
        return self.port_display
